analyzers/src/SonarAnalyzer.CSharp/Rules/UnconditionalJumpStatement.cs

```diff
@@ -75,15 +74,34 @@ private class LoopWalker : LoopWalkerBase<StatementSyntax, SyntaxKind>
                 SyntaxKind.CatchClause
             };
 
-            public LoopWalker(SyntaxNodeAnalysisContext context, ISet<SyntaxKind> loopStatements)
-                : base(context, loopStatements)
+            public LoopWalker(SyntaxNodeAnalysisContext context, ISet<SyntaxKind> loopStatements) : base(context, loopStatements) { }
+            public override void Visit()
             {
+                var csWalker = new CsLoopwalker(this);
+                csWalker.SafeVisit(rootExpression);
             }
 
-            public override void Visit()
+            protected override bool IsPropertyAccess(StatementSyntax node) =>
+                node.DescendantNodes().Any(x => x.IsKind(SyntaxKind.IdentifierName)
+                                                && x is IdentifierNameSyntax identifier
+                                                && semanticModel.GetSymbolInfo(identifier).Symbol is { } symbol
```

Comment:
We don't need `IsKind` check before `is IdentifierNameSyntax` in this case. That can simplify the syntax like this:
```suggestion
                node.DescendantNodes().OfType<IdentifierNameSyntax>().Any(x => semanticModel.GetSymbolInfo(x).Symbol is { } symbol
```